metrics/web/docs/features/deep_links/01_deep_linking.md

```diff
@@ -0,0 +1,106 @@
+
+The analysis stage aims the following goals:
+1. Define the detailed requirements for the deep linking feature;
+2. Investigate existing approaches for the deep linking in the Flutter Web applications.
+3. Define which approach satisfy our requirements: existing external solutions, or the custom-developed approach.
+
+# Requirements
+> Define requirements and make sure that they are complete.
+
+The deep links implementation approach that we are to select must satisfy the following requirements:
+- Possibility to handle deep links and restore the application's state using them;
+- Possibility to reflect the application's state in deep links as a response to UI events (e.g. filtering, dropdown selections, searching, etc.);
+
+# Prerequisites
+Before deep-diving into deep linking feature analysis, make sure to get familiar with the Flutter new navigation system ([Navigator 2.0](https://flutter.dev/docs/release/breaking-changes/route-navigator-refactoring)) which is [used in the Metrics Web application](https://github.com/Flank/flank-dashboard/blob/master/metrics/web/docs/features/navigation/01_navigation_design.md). The understanding of the Navigator 2.0 approach is important for the understanding deep linking feature.
+
+# Feasibility study
+> A preliminary study of the feasibility of implementing this feature.
+
+The `Deep Linking` feature improves the overall user experience in terms of navigation, so this feature makes sense for the project.
+
+To be able to implement the `Deep Linking` feature we should be able to perform the following actions:
+- Parsing the `query parameters` from the URL;
+- Updating the URL and the `query parameters` in response to any events.
+
+Since `Navigator 2.0` provides the navigation system that allows performing the actions listed above, and the Metrics Web application uses `Navigator 2.0`, it is possible to implement the `Deep Linking` feature in the Metrics Web application.
+
+# Landscape
+> Look for existing solutions in the area.
+
+There are 2 main approaches for implementing the `Deep Linking` feature in the Metrics Web application:
+- Using 3-rd party package (such as [`routemaster`](https://pub.dev/packages/routemaster), [`beamer`](https://pub.dev/packages/beamer), etc.);
+- A custom implementation based on the existing `Navigator 2.0` integration.
+
+Let's compare the listed approaches in a bit more detail.
+
+## 3-rd party package approach
+Consider the following pros and cons of the 3-rd party package utilization approach:
+
+Pros:
+- Provides ready-to-use implementation out of the box, including deep linking, nested navigation, and many other features.
+
+Cons:
+- May require code generation;
+- Requires the navigation system reintegration - meaning that we must use the 3-rd party `RouteInformationParser`s and `RouterDelegate`s.
+
+## Custom implementation
```

Comment:
Would it be useful to explain that the `custom implementation` means that we are going to implement the deep-linking using the `Navigator 2.0` features?